apps/app/repository.tf

```diff
@@ -77,32 +81,35 @@ resource "github_team_members" "contributors" {
 }
 
 resource "github_team_repository" "contributors_repo_permissions" {
-  team_id    = github_team.contributors.id
-  repository = github_repository.app.name
+  count      = local.create_repo ? 1 : 0
+  team_id    = github_team.contributors[0].id
+  repository = github_repository.app[0].name
   permission = "push"
 }
 
-resource "github_branch" "deployment" {
-  repository = github_repository.app.name
-  branch     = var.environment
+resource "github_branch" "all" {
+  for_each   = local.branches_and_envs
+  repository = github_repository.app[0].name
+  branch     = each.key
 }
 
 resource "github_branch_protection" "deployment" {
-  repository_id       = github_repository.app.name
-  pattern             = var.environment
+  for_each            = local.branches_and_envs
+  repository_id       = github_repository.app[0].name
+  pattern             = each.value
   allows_deletions    = false
   allows_force_pushes = false
 
   required_status_checks {
-    strict   = true
+    strict   = var.accesses_real_data
     contexts = ["Lint"]
   }
 
   required_pull_request_reviews {
-    dismiss_stale_reviews           = var.app_config.managed_repo.dismiss_stale_reviews
-    restrict_dismissals             = true
-    require_code_owner_reviews      = true
-    required_approving_review_count = var.app_config.managed_repo.num_of_approvals
+    dismiss_stale_reviews           = var.accesses_real_data
+    restrict_dismissals             = !var.accesses_real_data
+    require_code_owner_reviews      = !var.accesses_real_data
```

Comment:
Not sure we should stipulate this, it should come from the config as per the design as customers might wish to enforce specific reviewers even before a real data environment